libs/experimental/langchain_experimental/comprehend_moderation/toxicity.py

```diff
@@ -134,76 +132,34 @@ def validate(
-                            ):
-                                toxicity_found = True
-                                break
-
-                if action == BaseModerationActions.ALLOW:
-                    if not toxicity_labels:
-                        warnings.warn(
-                            "You have allowed toxic content without specifying "
-                            "any toxicity labels."
-                        )
-                    else:
-                        for item in response["ResultList"]:
-                            for label in item["Labels"]:
-                                if (
-                                    label["Name"] in toxicity_labels
-                                    and label["Score"] >= threshold
-                                ):
-                                    toxicity_found = True
-                                    break
-
-                if self.callback and self.callback.toxicity_callback:
-                    if toxicity_found:
-                        self.moderation_beacon["moderation_status"] = "LABELS_FOUND"
-                    asyncio.create_task(
-                        self.callback.on_after_toxicity(
-                            self.moderation_beacon, self.unique_id
-                        )
-                    )
-                if toxicity_found:
-                    raise ModerationToxicityError
+            toxicity_found = False
+            threshold = config.get("threshold")
+            toxicity_labels = config.get("labels")
+
+            if not toxicity_labels:
+                for item in response["ResultList"]:
+                    for label in item["Labels"]:
+                        if label["Score"] >= threshold:
+                            toxicity_found = True
+                            break
             else:
-                if response["ResultList"]:
-                    detected_toxic_labels = list()
-                    for item in response["ResultList"]:
-                        detected_toxic_labels.extend(item["Labels"])
-                    if any(item["Score"] >= 0.5 for item in detected_toxic_labels):
-                        if self.callback and self.callback.toxicity_callback:
-                            self.moderation_beacon["moderation_status"] = "LABELS_FOUND"
-                            asyncio.create_task(
-                                self.callback.on_after_toxicity(
-                                    self.moderation_beacon, self.unique_id
-                                )
-                            )
-                        raise ModerationToxicityError
+                for item in response["ResultList"]:
+                    for label in item["Labels"]:
+                        if (
+                            label["Name"] in toxicity_labels
+                            and label["Score"] >= threshold
+                        ):
+                            toxicity_found = True
```

Comment:
See the comment above.

Reply:
Will trace it in our next release and make changes accordingly